huaweicloud/services/iec/data_source_huaweicloud_iec_security_group.go

```diff
@@ -79,38 +81,38 @@ func dataSourceIECSecurityGroup() *schema.Resource {
 	}
 }
 
-func dataSourceIECSecurityGroupRead(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	iecClient, err := config.IECV1Client(GetRegion(d, config))
+func dataSourceIECSecurityGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	iecClient, err := cfg.IECV1Client(cfg.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud IEC client: %s", err)
+		return diag.Errorf("error creating IEC client: %s", err)
 	}
 
 	allSecGroups, err := groups.List(iecClient, nil).Extract()
 	if err != nil {
-		return fmtp.Errorf("Unable to retrieve security groups: %s", err)
+		return diag.Errorf("unable to retrieve security groups: %s", err)
 	}
 
 	total := len(allSecGroups.SecurityGroups)
 	if total < 1 {
-		return fmtp.Errorf("Your query returned no results")
+		return diag.Errorf("your query returned no results")
 	}
 
 	// filter security groups by name
 	var groupItem *groups.RespSecurityGroupEntity
 	name := d.Get("name").(string)
 	for _, group := range allSecGroups.SecurityGroups {
 		if group.Name == name {
```

Comment:
it's better to writing like this:
```suggestion
	for i := range allSecGroups.SecurityGroups {
		group := allSecGroups.SecurityGroups[i]
		if group.Name == name {
```